lute/std/libs/path/init.luau

```diff
@@ -0,0 +1,96 @@
+local system = require("@lute/system")
+
+local posix = require("@self/posix")
+local win32 = require("@self/win32")
+
+export type pathobj = posix.path | win32.path
```

Comment:
```suggestion
export type path = posix.path | win32.path
```

I don't think we should be abbreviating in names, and I don't understand what `object` actually adds for us here.